merkledag/merkledag.go

```diff
@@ -377,18 +377,39 @@ func EnumerateChildren(ctx context.Context, ds LinkService, root *cid.Cid, visit
 	} else if err != nil {
 		return err
 	}
+	v, _ := ctx.Value("progress").(*ProgressTracker)
```

Comment:
If the context does not have a progress tracker in it, I think this may panic (type asserting a nil value?)